modules/gitops-argocd-agent-install-principal-component.adoc

```diff
@@ -0,0 +1,314 @@
+----
+NAME                                            TYPE      PORT(S)
+openshift-gitops-agent-principal                ClusterIP 443/TCP
+openshift-gitops-agent-principal-healthz        ClusterIP 8030/TCP
+openshift-gitops-agent-principal-metrics        ClusterIP 9000/TCP
+openshift-gitops-agent-principal-redisproxy     ClusterIP 6379/TCP
+openshift-gitops-agent-principal-resource-proxy ClusterIP 9090/TCP
+----
+
+.. By default, the Operator creates a route for the Principal, which will be used to access Principal services. Verify the default route for the Principal component by running the following command:
++
+[source,terminal]
+----
+$ oc get route -n "<PRINCIPAL_NAMESPACE>" --context "<PRINCIPAL_CONTEXT>"
+----
++
+As a result, the following output is displayed:
++
+[source,terminal]
+----
+NAME                               HOST/PORT
+openshift-gitops-agent-principal   <HOST_NAME>
+----
+
+.. If you want to disable the route or use a `LoadBalancer` service, update the following section in the Argo CD resource:
++
+[source,yaml]
+----
+server:
+  service:
+    type: LoadBalancer        
+  route:
+    enabled: false
+----
+
+. Create the required namespaces.
+You must create one namespace for each Agent that you plan to connect to the Principal. The namespace names must match the Agent names. Run the following commands:
++
+[source,terminal]
+----
+$ oc create namespace agent-managed
+$ oc create namespace agent-autonomous
+----
++
+Each Agent requires a dedicated namespace because Argo CD Agent architecture uses a *namespace-to-agent* mapping model.  
+For managed Agents, the Principal sends applications only to those created in the corresponding Agent namespace on the hub cluster. Similarly, for autonomous Agents, applications are created in the respective Agent namespace at the hub.
+
+. Ensure that the `AppProject` resource used for your Argo CD Applications on the hub includes all Agent namespaces under the `spec.sourceNamespaces` field, for example:
++
+[source,yaml]
+----
+spec:
+  sourceNamespaces:
+  - "agent-managed"
+  - "agent-autonomous"
+----
++
+If you modify the `AppProject`, restart the Argo CD pods to apply the changes.
+
+. Create Required Secrets.
+The principal component uses a Public Key Infrastructure (PKI) to secure all communications.
++
+[NOTE]
+====
+The CLI-generated PKI is intended for development and testing purposes only. For production environments, use certificates issued by your organization's PKI or a trusted certificate authority.
+====
+
+.. Initialize the PKI
+
+... To initialize the Certificate Authority (CA) that signs other certificates, run the following command:
++
+[source,terminal]
+----
+$ argocd-agentctl pki init \
+     --principal-namespace "<PRINCIPAL_NAMESPACE>" \
+     --principal-context "<PRINCIPAL_CONTEXT>"
+----
++
+where:
+
+`<f>`:: Overwrites the existing CA if one already exists.
+`<principal-namespace>`:: Specifies the namespace where the Principal component is installed.
```

Comment:
since we already mentioned about principal-namespace, principal-context, is explanation of these fields  still required here?

Reply:
@jgwest, WDYT?

Reply:
Yes that's fine